desk/app/groups.hoon

```diff
@@ -1496,8 +1459,37 @@
     ::
         %del
       =.  cabals.group  (~(del by cabals.group) sect)
-      =.  cor  (verify-group-cabals [flag ~ group] cor)
-      go-core
+      =/  old-sect=(set sect:g)  (sy sect ~)
+      =.  go-core
+        ::  remove from members as needed
+        ::
+        %-  go-fleet-del-sects
+        :_  old-sect
+        %+  roll
+          ~(tap by fleet.group)
+        |=  [[s=ship =vessel:fleet:g] ships=(set ship)]
+        ?.  (~(has in sects.vessel) sect)  ships
+        (~(put in ships) s)
+      =^  core=_go-core  cor
+        %+  roll
+          ~(tap by channels.group)
+        |=  [[=nest:g =channel:g] [gc=_go-core cr=_cor]]
+        ::  repair readers as needed
+        ::
+        =.  gc  (go-channel-del-sects:gc nest old-sect)
+        ::  repair writers as needed
+        ::
+        =+  .^(has=? %gu (channel-scry nest))
+        ?.  has  [gc cr]
+        :: unsupported channel
+        ?.  ?=(?(%chat %heap %diary) p.nest)  [gc cr]
+        :: not host
+        ?:  !=(our.bowl p.q.nest)  [gc cr]
+        =/  cmd=c-channels:d  [%channel nest %del-writers old-sect]
+        =/  cage  [%channel-command !>(cmd)]
+        :-  gc
+        cr(cards [[%pass /groups/role %agent [p.q.nest %channels-server] %poke cage] cards.cr])
+      core
```

Comment:
The double context editing here kinda scares me. I guess you need them both explicitly because you can't do edits on ""`cor.gc`""? In that case, I'd probably prefer just inlining the loop. Lets you avoid doing manual `emit`s also.

(And given that scries have some non-trivial overhead I'd do that as late as reasonable.)

```suggestion
      =/  channels
        ~(tap by channels.group)
      |-
      ?~  channels  go-core
      =*  nest     p.i.channels
      =*  channel  q.i.channels
      ::  delete sects from channel readers
      ::
      =.  go-core  (go-channel-del-sects:gc nest old-sect)
      ::TODO  delete sects from channel writers?
      ::  unsupported channel type
      ::
      ?.  ?=(?(%chat %heap %diary) p.nest)  gc
      ::  we are not the host, so not responsible for role management
      ::
      ?:  !=(our.bowl p.q.nest)  gc
      ::  if the channel still exists in the server,
      ::  tell it about the role change
      ::
      =+  .^(has=? %gu (channel-scry nest))
      ?.  has  go-core
      =/  cmd=c-channels:d  [%channel nest %del-writers old-sect]
      =/  cage  [%channel-command !>(cmd)]
      =.  cor
        %-  emit
        [%pass /groups/role %agent [p.q.nest %channels-server] %poke cage]
      $(channels t.channels)
```